### PR TITLE
macOS: use Apple GL/GLU by default

### PR DESCRIPTION
**Pull request by adamjstewart**
Allowed me to successfully install glew. We should use these by default, I haven't found any other GL/GLU provider that works on macOS.

@aphecetche

### CODE REVIEW

**Comment by luca-heltai** [comment]
Are these available if one does not install XCode, but just the command line tools?

**Comment by luca-heltai** [comment]
On my system, for example, I don't have XCode, but the framework is available under `/Library/Developer/CommandLineTools/SDSKs/MacOSX13.1.sdk/System/Library/Frameworks/OpenGL.framework/`

**Comment by aphecetche** [comment]
@luca-heltai I don't know what's @adamjstewart take on this, but in my experience, trying to just work with the command line tools always lead me to subtle issues. So in the last few years I've ran into the habit of simply installing Xcode and be happy like that ;-)

**Review by aphecetche — DISMISSED** [review]
LGTM.

I guess the support (if deemed necessary/advisable) of the same framework but from the command line tools (instead of full Xcode) could be made in another PR.

**Comment by adamjstewart** [comment]
Yes, I'm not sure how to handle both the full Xcode suite and the command-line tools subset. When using the full suite, the command-line tools directory does not exist, and vice versa. Most things in Spack should work with either, although packages like Qt require the full suite.

Let me ask around on Slack and see if anyone has any ideas.

**Comment by adamjstewart** [comment]
One idea would be to encode these paths directly in the `package.py` so that the "prefix" provided in `packages.yaml` is ignored and the package searches both system locations and uses whichever exists or errors out if it can't find anything. Thoughts on this approach?

**Comment by adamjstewart** [comment]
In my testing, this also works with Xcode command line tools, which will always be present. Let's use that instead.

File: etc/spack/defaults/darwin/packages.yaml [hunk 1]
```diff
@@ -23,8 +23,20 @@ packages:
...
+    buildable: false
+    externals:
+    - spec: apple-gl@4.1.0
+      prefix: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
+  apple-glu:
+    buildable: false
+    externals:
+    - spec: apple-glu@1.3.0
+      prefix: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
   apple-libunwind:
```
Comment (adamjstewart) [inline etc/spack/defaults/darwin/packages.yaml:40]: Should we change the directory of libunwind too?

File: var/spack/repos/builtin/packages/apple-gl/package.py [hunk 2]
```diff
@@ -44,16 +39,14 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         # - for the rest of the build systems we'll assume that
         # setting the C_INCLUDE_PATH will be enough for the compilation phase
         # and *** for the link phase.
-        env.prepend_path("C_INCLUDE_PATH", self.sdk_base)
+        env.prepend_path("C_INCLUDE_PATH", self.prefix[:-4])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/apple-gl/package.py:42]: Why didn't this include the `.sdk` suffix, or even `.sdk/usr/include`?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/apple-gl/package.py:42]: @aphecetche?

**Review by kwryankrattiger — APPROVED** [review]
LGTM, thanks for moving this forward @adamjstewart !

**Pull request merged**